elfo-utils/src/lib.rs

```diff
@@ -11,6 +11,35 @@ pub mod time;
 
 pub use crossbeam_utils::CachePadded;
 
+/// Assert that expression matches the pattern. Optionally,
+/// part of the pattern could be returned:
+/// ```
+/// use elfo_utils::assert_matches;
+///
+/// assert_matches!(1, 1);
+/// let x = assert_matches!((1, 2), x = (x, _));
+/// assert_eq!(x, 1);
+/// ```
+#[macro_export]
+macro_rules! assert_matches {
```

Comment:
1. `elfo-utils` for very basic code used in multiple elfo crates
2. Don't see a reason for such sugar now